lib/webhooks/register.ts

```diff
@@ -140,30 +140,40 @@ function buildHandlerFromNode(edge: WebhookCheckResponseNode): WebhookHandler {
   switch (endpoint.__typename) {
     case 'WebhookHttpEndpoint':
       handler = {
-        id: edge.node.id,
         deliveryMethod: DeliveryMethod.Http,
+        privateMetafieldNamespaces: edge.node.privateMetafieldNamespaces,
         callbackUrl: endpoint.callbackUrl,
         // This is a dummy for now because we don't really care about it
         callback: async () => {},
       };
       break;
     case 'WebhookEventBridgeEndpoint':
       handler = {
-        id: edge.node.id,
         deliveryMethod: DeliveryMethod.EventBridge,
         arn: endpoint.arn,
       };
       break;
     case 'WebhookPubSubEndpoint':
       handler = {
-        id: edge.node.id,
         deliveryMethod: DeliveryMethod.PubSub,
         pubSubProject: endpoint.pubSubProject,
         pubSubTopic: endpoint.pubSubTopic,
       };
       break;
   }
 
+  // Set common fields
+  handler.id = edge.node.id;
+  handler.includeFields = edge.node.includeFields;
+  handler.metafieldNamespaces = edge.node.metafieldNamespaces;
+
+  // Sort the array fields to make them cheaper to compare later on
+  handler.includeFields?.sort();
+  handler.metafieldNamespaces?.sort();
+  if (handler.deliveryMethod === DeliveryMethod.Http) {
```

Comment:
could we add a comment explaining why `privateMetafieldNamespaces` needs this condition?